File: frontend/src/services/baggageService.js
import api from './api'

const baggageService = {
  // Get all baggage
  getAllBaggage: async () => {
    const response = await api.get('/api/v1/baggage/all')
    return response.data
  },

  // Get baggage by ID
  getBaggageById: async (id) => {
    const response = await api.get(`/api/v1/baggage/${id}`)
    return response.data
  },

  // Get baggage by tag code
  getBaggageByTag: async (tagCode) => {
    const response = await api.get(`/api/v1/baggage/tag/${tagCode}`)
    return response.data
  },

  // Get baggage by ticket
  getBaggageByTicket: async (ticketId) => {
    const response = await api.get(`/api/v1/baggage/ticket/${ticketId}`)
    return response.data
  },

  // Get baggage by trip
  getBaggageByTrip: async (tripId) => {
    const response = await api.get(`/api/v1/baggage/trip/${tripId}`)
    return response.data
  },

  // Calculate baggage fee
  calculateBaggageFee: async (weightKg) => {
    const response = await api.get('/api/v1/baggage/calculate-fee', {
      params: { weightKg },
    })
    return response.data
  },

  // Create baggage (CLERK, ADMIN)
  createBaggage: async (data) => {
    const response = await api.post('/api/v1/baggage/create', data)
    return response.data
  },

  // Update baggage (CLERK, ADMIN)
  updateBaggage: async (id, data) => {
    const response = await api.put(`/api/v1/baggage/update/${id}`, data)
    return response.data
  },

  // Delete baggage (ADMIN)
  deleteBaggage: async (id) => {
    await api.delete(`/api/v1/baggage/delete/${id}`)
  },
}

export default baggageService

File: src/main/java/com/bers/services/service/BaggageService.java
package com.bers.services.service;

import com.bers.api.dtos.BaggageDtos.BaggageCreateRequest;
import com.bers.api.dtos.BaggageDtos.BaggageResponse;
import com.bers.api.dtos.BaggageDtos.BaggageUpdateRequest;

import java.math.BigDecimal;
import java.util.List;

public interface BaggageService {

    BaggageResponse createBaggage(BaggageCreateRequest request);

    BaggageResponse updateBaggage(Long id, BaggageUpdateRequest request);

    BaggageResponse getBaggageById(Long id);

    BaggageResponse getBaggageByTagCode(String tagCode);

    List<BaggageResponse> getAllBaggage();

    List<BaggageResponse> getBaggageByTicketId(Long ticketId);

    List<BaggageResponse> getBaggageByTripId(Long tripId);

    void deleteBaggage(Long id);

    BigDecimal calculateBaggageFee(BigDecimal weightKg);

    BigDecimal getTotalWeightByTrip(Long tripId);
}
